lib/grpc/GrpcService.ts

```diff
@@ -77,19 +78,28 @@ const createPlaceOrderResponse = (result: PlaceOrderResult) => {
  * Creates an xudrpc PlaceOrderEvent message from a [[PlaceOrderEvent]].
  */
 const createPlaceOrderEvent = (e: PlaceOrderEvent) => {
-  const response = new xudrpc.PlaceOrderEvent();
-  switch (e.case) {
-    case PlaceOrderEventCase.InternalMatch:
-      response.setInternalMatch(createOrder(e.payload as Order));
+  const placeOrderEvent = new xudrpc.PlaceOrderEvent();
+  switch (e.type) {
+    case PlaceOrderEventType.InternalMatch:
+      placeOrderEvent.setInternalMatch(createOrder(e.payload as Order));
       break;
-    case PlaceOrderEventCase.SwapResult:
-      response.setSwapResult(createSwapResult(e.payload as SwapResult));
+    case PlaceOrderEventType.SwapResult:
+      placeOrderEvent.setSwapResult(createSwapResult(e.payload as SwapResult));
       break;
-    case PlaceOrderEventCase.RemainingOrder:
-      response.setRemainingOrder(createOrder(e.payload as Order));
+    case PlaceOrderEventType.RemainingOrder:
+      placeOrderEvent.setRemainingOrder(createOrder(e.payload as Order));
+      break;
+    case PlaceOrderEventType.FailedSwap:
```

Comment:
perhaps extract the following object construction, to match the other cases

Reply:
I didn't originally because it's code that's not being reused anywhere else, but I can extract it for consistency.